### PR TITLE
542-add warning message if some accounts are corrupted

### PR DESCRIPTION
**Pull request by Pierregvx**

### CODE REVIEW

File: internal/handler/wallet/getall.go [hunk 1]
```diff
@@ -41,12 +41,35 @@ func (h *walletGetAll) Handle(params operations.AccountListParams) middleware.Re
 			})
 	}
 
-	for i := 0; i < len(wallets); i++ {
-		modelWallet := createModelWallet(wallets[i])
+	for i := 0; i < len(walletsWithoutError); i++ {
+		status := "ok"
```
Comment (Thykof) [inline internal/handler/wallet/getall.go:45]: we should use a constant variable, the same as the one we use when creating a new account

File: internal/handler/wallet/getall.go [hunk 2]
```diff
@@ -41,12 +41,35 @@ func (h *walletGetAll) Handle(params operations.AccountListParams) middleware.Re
...
+		modelWallet := createModelWallet(walletsWithoutError[i])
 		modelWallet.CandidateBalance = models.Amount(fmt.Sprint(infos[i].CandidateBalance))
 		modelWallet.Balance = models.Amount(fmt.Sprint(infos[i].Balance))
+		modelWallet.Status = &status
 		wlts = append(wlts, &modelWallet)
 	}
+	for u := 0; u < len(walletsWithError); u++ {
+		status := "corrupted"
+		modelWalletErr := createModelWallet(walletsWithError[u])
+		modelWalletErr.Status = &status
```
Comment (Thykof) [inline internal/handler/wallet/getall.go:55]: can we modify `createModelWallet` instead of setting the Status here?

File: pkg/wallet/wallet.go [hunk 3]
```diff
@@ -307,16 +308,23 @@ func LoadFile(filePath string) (Wallet, *WalletError) {
...
-		return Wallet{}, errMissingFields
+		return baseAccount, errMissingFields
 	}
 
 	account, err := accountSerialized.toAccount()
 	if err != nil {
-		return Wallet{}, &WalletError{fmt.Errorf("deserializing account '%s': %w", filePath, err), utils.ErrAccountFile}
+		return baseAccount, &WalletError{fmt.Errorf("deserializing account '%s': %w", filePath, err), utils.ErrAccountFile}
 	}
+	account.Status = "ok"
```
Comment (Thykof) [inline pkg/wallet/wallet.go:327]: create a constant variable for "ok"

File: pkg/wallet/wallet.go [hunk 4]
```diff
@@ -307,16 +308,23 @@ func LoadFile(filePath string) (Wallet, *WalletError) {
 	if err != nil {
 		return Wallet{}, &WalletError{fmt.Errorf("unmarshalling file '%s': %w", filePath, err), utils.ErrAccountFile}
 	}
-
+	walletHasNickname := len(accountSerialized.Nickname) > 0
+	baseAccount := Wallet{
+		Nickname: "unknown wallet",
```
Comment (Thykof) [inline pkg/wallet/wallet.go:313]: is it possible to get the wallet nickname from the filename?

File: web-frontend/src/pages/Index/Index.tsx [hunk 5]
```diff
@@ -1,28 +1,45 @@
...
-  const { error, data, isLoading } = useResource<AccountObject[]>('accounts');
+  const { error, okAccounts, corruptedAccounts, isLoading } = fetchAccounts();
+  const corruptedAccountsNames = corruptedAccounts?.map(
+    (account: AccountObject) => account.nickname,
+  );
+  const accountsStringified =
+    corruptedAccountsNames
+      ?.map(
+        (account: string, index: number) =>
+          `${account}${
```
Comment (Thykof) [inline web-frontend/src/pages/Index/Index.tsx:21]:
```suggestion
        (name: string, index: number) =>
          `${name}${
```

File: plugin.zip
Comment (Thykof) [inline plugin.zip:1]: remove this

**Review by Thykof — CHANGES_REQUESTED** [review]

File: pkg/wallet/wallet.go (hunk 3 above)
Comment (Thykof) [reply pkg/wallet/wallet.go:332]: now you can use it!

File: pkg/wallet/wallet.go [hunk 6]
```diff
@@ -307,16 +310,26 @@ func LoadFile(filePath string) (Wallet, *WalletError) {
 	if err != nil {
 		return Wallet{}, &WalletError{fmt.Errorf("unmarshalling file '%s': %w", filePath, err), utils.ErrAccountFile}
 	}
-
+	walletHasNickname := len(accountSerialized.Nickname) > 0
+	_, nicknameFromFileName := filepath.Split(filePath)
+	nicknameFromFileName = strings.TrimPrefix(nicknameFromFileName, "wallet_")
+	nicknameFromFileName = strings.TrimSuffix(nicknameFromFileName, ".yaml")
```
Comment (Thykof) [inline pkg/wallet/wallet.go:316]: you can extract this into a function near https://github.com/massalabs/station-massa-wallet/blob/86040bea9ca6414c2087770de7968e2b08548028/pkg/wallet/wallet.go#L389

**Review by Thykof — CHANGES_REQUESTED** [review]

File: pkg/wallet/wallet.go (hunk 6 above)
Comment (Thykof) [reply pkg/wallet/wallet.go:316]:
```go
```

Comment (Thykof) [reply pkg/wallet/wallet.go]: you can create this function and put it bellow that function: https://github.com/massalabs/station-massa-wallet/blob/86040bea9ca6414c2087770de7968e2b08548028/pkg/wallet/wallet.go#L389

```go
func NicknameFromFilePath(filePath string): string {
	_, nicknameFromFileName := filepath.Split(filePath)
	nicknameFromFileName = strings.TrimPrefix(nicknameFromFileName, "wallet_")
	nicknameFromFileName = strings.TrimSuffix(nicknameFromFileName, ".yaml")

	return nicknameFromFileName
}
```

Comment (Thykof) [reply pkg/wallet/wallet.go]: in golang, we usually don't prefix the getter functions with `get`

**Review by Thykof — APPROVED** [review]

**Review by pivilartisant — APPROVED** [review]

**Pull request merged**